mautic/mautic
# Elastic Email integration

| Q  | A
| --- | ---
| Bug fix? |  n
| New feature? | y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/126
| Related developer documentation PR URL | n
| Issues addressed (#s or URLs) | n
| BC breaks? | n
| Deprecations? |  n

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Elastic Email offer 150 000 emails/month free. This PR added support to Mautic with SMTP sending and  bounced&underliverd messages.

#### Steps to test send email:
1. Create Elastic Email account. Go to Settings > SMTP/API tab and copy SMTP Configuration. Also your sending domain in Settings > Domains (for test you don' need validate domain by DNS).
2. Set up that credits to Mautic > Configuration > Emails tab
3. Send test email

#### Steps to test bounced email
1.  Go to Elastic Email account > Notifications
2. Fill Notification url with http://your-mautic-url.tld/mailer/elasticemail/callback
3. Check from Notify On: Unsubscribed, Complaints, Bounce/Error
4. Create contact with non exist email address and send email to this contact
5. After few minutes check contact profile
6. Works If is marked as bounce

Support for these statuses: 'AbuseReport', 'Unsubscribed', 'NotDelivered','NoMailbox', 'AccountProblem', 'DNSProblem', 'Unknown', 'Spam'

![notification](https://cloud.githubusercontent.com/assets/462477/20074338/1e28e494-a530-11e6-97a8-4e2c5e02b096.PNG)
